Lib/lib2to3/tests/test_refactor.py

```diff
@@ -206,6 +206,23 @@ def read_file():
         self.assertNotEqual(old_contents, new_contents)
         return new_contents
 
+    def refactor_file(self, test_file, fixers=_2TO3_FIXERS):
+        tmpdir = tempfile.mkdtemp(prefix="2to3-test_refactor")
+        self.addCleanup(shutil.rmtree, tmpdir)
+        shutil.copy(test_file, tmpdir)
+        test_file = os.path.join(tmpdir, os.path.basename(test_file))
+        os.chmod(test_file, 0o644)
+
+        def read_file():
+            with open(test_file, "rb") as fp:
+                return fp.read()
+
+        old_contents = read_file()
+        rt = self.rt(fixers=fixers)
```

Comment:
Would be nicer if you extracted lines 210-221 to a separate function that both `check_file_refactoring()` and `refactor_file()` would reuse. Now there's copypasta.